vendors/infineon/boards/xmc4800_iotkit/ports/wifi/iot_wifi.c

```diff
@@ -415,7 +416,7 @@ WIFIReturnCode_t WIFI_Ping( uint8_t * pucIPAddr,
     {
 	  uint32_t time;
 
-      if (esp_ping(host_name, &time, 1) != espOK)
+      if ( (ret = esp_ping(host_name, &time, 1) ) != espOK)
```

Comment:
```if ( (a=b) != c )``` functionally correct, but preferably not. I don't have a most up-to-date MISHRA pdf at hand, but ```MISRA C:2004, 13.1 - Assignment operators shall not be used in expressions that yield a Boolean value```. This rule is "required".